### PR TITLE
Implement unit tests for each command

### PR DESCRIPTION
**Pull request by johnallen3d**
While writing these tests the opportunity was taken to refactor the Ruby
implementation to be a bit more D.R.Y. This was done given the relative
safety of having the integration tests.

### CODE REVIEW

**Comment by technekes-bundler** [comment]
:eyes: @jfinney122

**Comment by johnallen3d** [comment]
This is a pretty massive change @jfinney122. Let me know if you'd like to take a few minutes to talk something through.

File: bin/nib [hunk 1]
```diff
@@ -39,8 +39,11 @@ end
 on_error do |exception|
   case exception
   when GLI::UnknownCommand
-    # delegate unknown commands over to `docker-compose`
-    system("docker-compose #{ARGV.join(' ')}")
+    begin
+      # delegate unknown commands over to `docker-compose`
+      system("docker-compose #{ARGV.join(' ')}")
+    rescue Interrupt
+    end
```
Comment (jfinney122) [inline bin/nib]: Is this code related to the PR?

File: spec/unit/code_climate_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+RSpec.describe Nib::CodeClimate do
+  subject { described_class.new(nil, []) }
+
+  before do
+    allow(subject).to receive(:system)
+  end
+
+  it 'runs the codeclimate docker image' do
+    expect(subject.script).to match(/docker-compose.*run.*--rm.*codeclimate/)
```
Comment (jfinney122) [inline spec/unit/code_climate_spec.rb]: do we want to make sure the command is getting passed to codeclimate? e.g. test with the analyze command?

**Comment by johnallen3d** [comment]
OK @jfinney122, I believe I've addressed your concerns.

**Comment by jfinney122** [comment]
@johnallen3d this looks good to me 👍  thanks for reviewing these changes with me
